pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/pubsub/WorkflowStatusUpdate.java

```diff
@@ -88,46 +90,56 @@ public static void updateCustomWorkflow(Map<String, String> attributesMap, Strin
             return;
         }
 
-        List<ParticipantData> participantDatas = participantDataDao.getParticipantDataByParticipantId(ddpParticipantId);
+        updateWorkflowData(ddpParticipantId, workflow, status, instance);
+
+        try {
+            if (isATStudy(studyGuid)) {
+                log.info("Generating default values for ATCP participant: {}, workflow: {}, status: {}", ddpParticipantId,
+                        workflow, status);
+                ATDefaultValues basicDefaultDataMaker = new ATDefaultValues();
+                basicDefaultDataMaker.generateDefaults(studyGuid, ddpParticipantId);
+            }
+        } catch (Exception e) {
+            throw new DsmInternalError("Could not add AT default values", e);
+        }
+    }
+
+    protected static void updateWorkflowData(String ddpParticipantId, String workflow, String status,
+                                             DDPInstance instance) {
+        String instanceName = instance.getName();
+        List<ParticipantData> participantDataList =
+                participantDataDao.getParticipantDataByParticipantId(ddpParticipantId);
         Optional<FieldSettingsDto> fieldSetting =
-                fieldSettingsDao.getFieldSettingByColumnNameAndInstanceId(Integer.parseInt(instance.getDdpInstanceId()), workflow);
+                fieldSettingsDao.getFieldSettingByColumnNameAndInstanceId(instance.getDdpInstanceIdAsInt(), workflow);
         if (fieldSetting.isEmpty()) {
-            throw new DsmInternalError(String.format("Invalid workflow name %s for instance %s", workflow, instance.getName()));
+            throw new DsmInternalError(String.format("Invalid workflow name %s for instance %s", workflow, instanceName));
         }
 
         FieldSettingsDto setting = fieldSetting.get();
         String fieldType = setting.getFieldType();
-        boolean isOldParticipant = participantDatas.stream().anyMatch(
+        boolean isOldParticipant = participantDataList.stream().anyMatch(
```

Comment:
What does "old participant" mean here?  If it's just a pre-existing participant vs. a soon-to-be-created one, maybe rename to `doesParticipantExist` because oldness could also mean something like "participant is of a certain age" or "this participant was first enrolled in datstat" (or something else).

Reply:
I can rename. This is legacy code I've been working on bit by bit for months. I try not to change too much, like variable names, to preserve some of the original code, but at this point I have rewritten this method a few times. (See the git history.)